onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu

```diff
@@ -185,10 +185,22 @@ Status LaunchSkipLayerNormKernel(
                                                  maybe2half<T>(epsilon), output,
                                                  skip_input_bias_add_output, hasBias, hasSkipInputBiasAdditionOutput);
     } else {
-      constexpr int block_size = 256;
-      SkipLayerNormKernel<T, block_size, Simplified>
-          <<<grid_size, block_size, 0, stream>>>(ld, input, skip, beta, gamma, bias,
-                                                 maybe2half<T>(epsilon), output, skip_input_bias_add_output);
+      int cuda_device_id{0};
+      int max_threads_per_block{1024};
+      CUDA_CALL_THROW(cudaGetDevice(&cuda_device_id));
+      CUDA_CALL_THROW(cudaDeviceGetAttribute(&max_threads_per_block, cudaDevAttrMaxThreadsPerBlock, cuda_device_id));
+
+      if (max_threads_per_block >= 1024 && ld > (512 * 4)) {
```

Comment:
You can pass max_threads_per_block from Attention class.

Reply:
Just keep current api.

Reply:
The function calls of cudaGetDevice and cudaDeviceGetAttribute takes time. There is no need to call them in each kernel launch if they can be done only once.